pkg/apis/pipeline/v1/pipelinerun_defaults.go

```diff
@@ -36,6 +36,10 @@ func (pr *PipelineRun) SetDefaults(ctx context.Context) {
 // SetDefaults implements apis.Defaultable
 func (prs *PipelineRunSpec) SetDefaults(ctx context.Context) {
 	cfg := config.FromContextOrDefaults(ctx)
+	if prs.Timeouts == nil {
+		prs.Timeouts = &TimeoutFields{}
+		prs.Timeouts.Pipeline = &metav1.Duration{Duration: time.Duration(cfg.Defaults.DefaultTimeoutMinutes) * time.Minute}
+	}
```

Comment:
seems like this is a separate change to pipelinerun.timeouts -- could this be in a separate commit?

Reply:
also we set prs.Timeouts.Pipeline twice - in line 41 and line 44

Reply:
Thanks Jerop. Yeah I will move it into a separate PR.

But to confirm on this addition, I might be mistaken but I wondered if the added condition would make sense if there are not any Timeouts specified. 

ie. previously the timeout default is set with PipelineRun.Timeout but it was not moved to v1 since timeout is deprecated.

Reply:
maybe this is what we need (replacing line 39 to line 45)
```go
	if prs.Timeouts == nil {
		prs.Timeouts = &TimeoutFields{}
	}
	if prs.Timeouts.Pipeline == nil {
		prs.Timeouts.Pipeline = &metav1.Duration{Duration: time.Duration(cfg.Defaults.DefaultTimeoutMinutes) * time.Minute}
	}
```

Reply:
Thanks Jerop, I edited it at https://github.com/tektoncd/pipeline/pull/6311